### PR TITLE
feat: add noise suppression audio plugin

### PR DESCRIPTION
**Pull request by A-KVerma**

### CODE REVIEW

File: src/views/footerView.jsx [hunk 1]
```diff
@@ -52,19 +56,48 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
...
   const params = useParams();
   const pluginRef = useRef(null);
+  const audiopluginRef = useRef(null);
   const isAllowedToPublish = useHMSStore(selectIsAllowedToPublish);
   const permissions = useHMSStore(selectPermissions);
   const [showEndRoomModal, setShowEndRoomModal] = useState(false);
   const [lockRoom, setLockRoom] = useState(false);
   const [showMenu, setShowMenu] = useState(false);
 
+  const noiseSupression = useHMSStore(
```
Comment (triptu) [inline src/views/footerView.jsx:66]: `isNoiseSupressionOn` boolean should start with is, has etc.

File: src/views/footerView.jsx [hunk 2]
```diff
@@ -52,19 +56,48 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
...
   const initialModalProps = {
     show: false,
     title: "",
     body: "",
   };
   const [errorModal, setErrorModal] = useState(initialModalProps);
 
+  function createNoiseSupresionPlugin() {
+    if (!audiopluginRef.current) {
+
```
Comment (triptu) [inline src/views/footerView.jsx:80]: why new line?

File: src/views/footerView.jsx [hunk 3]
```diff
@@ -52,19 +56,48 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
...
+      audiopluginRef.current = new HMSNoiseSuppressionPlugin();
+    }
+  }
+
+  async function startAudioPlugin() {
```
Comment (triptu) [inline src/views/footerView.jsx:85]: can we make this - `addNoiseSuppressionPlugin` and `removeNoiseSuppressionPlugin`, similar for vb functions as well to make these more clear for people going through the sample app.

File: src/views/footerView.jsx [hunk 4]
```diff
@@ -85,10 +118,13 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
       pluginRef.current = null;
     }
   }
-
   function handleVirtualBackground() {
     isVBPresent ? removePlugin() : startPlugin();
   }
+  function handleNoiseSupression() {
+   noiseSupression ? removeAudioPlugin() : startAudioPlugin();
+}
```
Comment (triptu) [inline src/views/footerView.jsx:126]: is prettier run on this?

File: src/views/footerView.jsx [hunk 5]
```diff
@@ -52,19 +56,48 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
...
+    if (!audiopluginRef.current) {
+
+      audiopluginRef.current = new HMSNoiseSuppressionPlugin();
+    }
+  }
+
+  async function startAudioPlugin() {
+      createNoiseSupresionPlugin();
+
+      audiopluginRef.current.setNoiseSuppression(!noiseSupression);
```
Comment (triptu) [inline src/views/footerView.jsx:88]: when would the input to setNoiseSuppression not be true? we can do `audiopluginRef.current.setNoiseSuppression(true)`. It also be true by default, this line shouldn't be required.

File: src/views/footerView.jsx [hunk 6]
```diff
@@ -330,6 +380,10 @@ export const ConferenceFooter = ({ isChatOpen, toggleChat }) => {
         backgroundButtonOnClick={handleVirtualBackground}
         isAudioMuted={!isLocalAudioEnabled}
         isVideoMuted={!isLocalVideoEnabled}
+         isBackgroundEnabled={isVBPresent}
+        classes={{
+          rightRoot: "flex",
```
Comment (triptu) [inline src/views/footerView.jsx:384]: what's this for?

**Comment by ashwins93** [comment]
Duplicate PR. Refer #369

**Pull request closed without merging**